zebra-consensus/src/block/subsidies.rs

```diff
@@ -0,0 +1,236 @@
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_founders_reward() -> Result<(), Report> {
+        let network = Network::Mainnet;
+        let canopy_height = Canopy.activation_height(network).unwrap();
+        let last_founder_reward = (canopy_height - 1).unwrap();
+
+        assert_eq!(0, founders_reward(Height(0), network));
+        assert_eq!(12500, founders_reward(Height(1), network));
+        assert_eq!(125000000, founders_reward(last_founder_reward, network));
+        assert_eq!(
+            0,
+            founders_reward((last_founder_reward + 1).unwrap(), network)
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_funding_stream() -> Result<(), Report> {
+        let network = Network::Mainnet;
+        let canopy_height = Canopy.activation_height(network).unwrap();
+        let last_founder_reward = (canopy_height - 1).unwrap();
+
+        assert_eq!(0, funding_stream(Height(0), network, fs::Receiver::ECC));
+        assert_eq!(0, funding_stream(Height(0), network, fs::Receiver::ZF));
+        assert_eq!(0, funding_stream(Height(0), network, fs::Receiver::MG));
+
+        assert_eq!(0, funding_stream(Height(1), network, fs::Receiver::ECC));
+        assert_eq!(0, funding_stream(Height(1), network, fs::Receiver::ZF));
+        assert_eq!(0, funding_stream(Height(1), network, fs::Receiver::MG));
+
+        assert_eq!(
+            0,
+            funding_stream(last_founder_reward, network, fs::Receiver::ECC)
+        );
+        assert_eq!(
+            0,
+            funding_stream(last_founder_reward, network, fs::Receiver::ZF)
+        );
+        assert_eq!(
+            0,
+            funding_stream(last_founder_reward, network, fs::Receiver::MG)
+        );
+
+        assert_eq!(
+            87500000,
+            funding_stream(
+                (last_founder_reward + 1).unwrap(),
+                network,
+                fs::Receiver::ECC
+            )
+        );
+        assert_eq!(
+            62500000,
+            funding_stream(
+                (last_founder_reward + 1).unwrap(),
+                network,
+                fs::Receiver::ZF
+            )
+        );
+        assert_eq!(
+            100000000,
+            funding_stream(
+                (last_founder_reward + 1).unwrap(),
+                network,
+                fs::Receiver::MG
+            )
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn miner_subsidy_test() -> Result<(), Report> {
+        let network = Network::Mainnet;
+        let canopy_height = Canopy.activation_height(network).unwrap();
+        let last_founder_reward = (canopy_height - 1).unwrap();
+
+        assert_eq!(0, miner_subsidy(Height(0), Network::Mainnet));
+        assert_eq!(50000, miner_subsidy(Height(1), Network::Mainnet));
+        assert_eq!(500000000, miner_subsidy(last_founder_reward, network));
+        assert_eq!(
+            1000000000,
+            miner_subsidy((last_founder_reward + 1).unwrap(), network)
+        );
```

Comment:
Testnet also hasn't halved yet. It halves at block height 1116000 (a different height to mainnet, because the block target spacing change in Blossom activated at a lower height on testnet). The current testnet height is ~1100580.

Reply:
Thanks, I've added a ticket for post-halving testnet blocks (#1104).